### PR TITLE
Use friendly name in POM for published artifacts - fixes #1396

### LINKED ISSUE #1396: incorrect <name> entry in pom ("core" instead of "cats-core")
if I do `sbt coreJVM/publishLocal` and inspect the resulting `cats-core_2.11.pom` I see:

```xml
    <name>core</name>
```

instead of the expected `cats-core`.

this then confused dbuild in the context of the Scala 2.12 community build, since the macro-compat project (attn @milessabin) also has a `org.typelevel#core` and the duplication caused dbuild to complain.

I am not a pom/maven/ivy jockey at all and I have basically zero idea what the implications of the `<name></name>` part of a pom are. but it seems plausible to me that this could cause other sorts of tooling type problems in other contexts.

I didn't check to see if other subprojects are affected.

after `set name in coreJVM := "cats-core"` or with

```diff
-  .settings(moduleName := "cats-core")
+  .settings(moduleName := "cats-core", name := "cats-core")
```

a correct pom is generated. what the difference between `moduleName` and `name` is I have no idea

### PR DESCRIPTION
**Pull request by adelbertc**
cc @SethTisue @BennyHill @travisbrown

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -274,7 +274,7 @@ lazy val lawsJS = laws.js
 
 lazy val free = crossProject.crossType(CrossType.Pure)
   .dependsOn(macros, core, tests % "test-internal -> test")
-  .settings(moduleName := "cats-free")
+  .settings(moduleName := "cats-free", name := "Cats Free")
```
Comment (travisbrown) [inline build.sbt]: Any reason "Free" is capitalized here (unlike the others)?

Comment (adelbertc) [inline build.sbt]: I just did it because it was named after the Free monad which I usually see capitalized. I can make it lower case if folks prefer that.

**Comment by SethTisue** [comment]
let's see what the community build thinks: https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/853/consoleFull (404 til Jenkins catches up)

File: build.sbt (hunk 1 above)
Comment (travisbrown) [inline build.sbt]: Fair enough. I don't have a strong opinion either way (but would prefer lowercase).

**Comment by SethTisue** [comment]
```
[cats] Resolving: https://github.com/adelbertc/cats.git#4522b6117574d24c41ce870a72ffc8257afd2e67 ...
...
[info] Project cats--------------------: SUCCESS (project rebuilt ok)
```

community build run is green, so this LGTM

**Comment by adelbertc** [comment]
@SethTisue Awesome possum

@travisbrown Is that a 👍 from you or would you prefer I change the capitalization?

**Comment by travisbrown** [comment]
Sure, as it is is fine. :+1:

**Comment by non** [comment]
Seems reasonable to me. Thanks @adelbertc! :+1:

**Pull request merged**